nemo_aligner/utils/distributed.py

```diff
@@ -320,3 +324,37 @@ def sync_and_consume_over_stored_time(self, name=""):
         yield from output_list
 
         del self.stored_results[name]
+
+
+@dataclass
+class Timer:
+    """Timer to tell us when the time limit has reached
+    """
+
+    duration: Optional[str]
+
+    def __post_init__(self):
+        self._duration = float("inf")
+
+        if isinstance(self.duration, str):
```

Comment:
I think this would be safer:
```suggestion
        if self.duration is not None:
```

Reply:
are you thinking that checking this way let's the users crash immediately if they didn't give a string? whereas if it's another dtype it'll crash during the training run

Reply:
I don't think it'll crash if for instance someone sets `max_time_per_run` to an integer value, it seems to me it'll just get ignored, right? (though it might crash in `exp_manager()`, not sure)

Reply:
not sure about exp_manager but yeah it won't crash. Are you saying this is safer because we are implicitly forcing the type to be string here?

Reply:
Yes -- you can assert it but my main concern was someone accidentally providing a number instead of a string and having it silently ignored (sorry I should have been clearer)